src/main/java/org/testng/TestRunner.java

```diff
@@ -183,12 +184,29 @@ protected TestRunner(IConfiguration configuration,
                     IAnnotationFinder finder,
                     boolean skipFailedInvocationCounts,
                     Collection<IInvokedMethodListener> invokedMethodListeners,
-                    List<IClassListener> classListeners, Comparator<ITestNGMethod> comparator) {
+                    List<IClassListener> classListeners,
+                    Comparator<ITestNGMethod> comparator) {
+    this(configuration, suite, test, outputDirectory, finder, skipFailedInvocationCounts,
+            invokedMethodListeners, classListeners, comparator,
+            Collections.<Class<? extends IDataProviderListener>, IDataProviderListener>emptyMap());
+  }
+
+  protected TestRunner(IConfiguration configuration,
```

Comment:
Oops.. sorry. This constructor can't be deprecated because its being used by org.testng.SuiteRunner.DefaultTestRunnerFactory#newTestRunner(org.testng.ISuite, org.testng.xml.XmlTest, java.util.Collection<org.testng.IInvokedMethodListener>, java.util.List<org.testng.IClassListener>, java.util.Map<java.lang.Class<? extends org.testng.IDataProviderListener>,org.testng.IDataProviderListener>)